### PR TITLE
[ACI] Updating summary/description fields to generate documentation

### PR DESCRIPTION
**Pull request by jluk**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: What do you mean by "base level" - host-level? Does the user need to know that, can they just input the OS type of the image they bring?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Currently always is the only option and defaulted, desc should update when r2c/never are implemented.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: We should decide on whether we want to call it "volume" or "share" AFAIK its interchangeable?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Throughout all references to a single container, I updated it to state "container instance".

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: On Line 709 - I don't think you can actually specify a UDP port today. We should remove this bit if we generate docs before it is supported.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Added summary fields on each explicit rest call to conform to how ACS gens docs.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Added each property of the container group to conform to how ACS describes each service.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Update 200 response descriptions to OK for consistency

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Is this added detail correct about not deleting the volumes associated and any IP addresses/etc. being in the same RG?

**Comment by jluk** [comment]
@yolo3301 could you review this PR/comments when you get a free moment? Once we improve on the descriptions/summaries a bit we can generate the docs for more user feedback.

**Review by jluk — COMMENTED** [review]
Leaving comments here for @yolo3301 to review description/summary updates.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (veronicagg) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Should be fine, as long as it's helpful for the customer.

**Review by veronicagg — CHANGES_REQUESTED** [review]
Changes look good to me. 
Per request from @jluk, PR is pending review from @yolo3301 . 
Once I get confirmation from you that the PR look good, I can approve and merge. Thanks!

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (veronicagg) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: You may want to address this warning: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/270502075#L673
"The summary and description values should not be same."

**Review by veronicagg — CHANGES_REQUESTED** [review]

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (mmacy) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Might be good to mention that an Azure file *share* is mounted as a volume.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (mmacy) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: storage account *access* key

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (mmacy) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Another volume vs. share comment, Azure Files does not have volumes, but shares. We definitely want to be careful not to mix terms. I'll refrain from further volume/share comments, you get the idea. :)

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (mmacy) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Must not contain colon (:).

(list full character name first, then the character in parens, no quotes)

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (mmacy) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Recommend simply striking "The type of IP address determining" from this, or go with something like:

"Specifies whether the IP is exposed to the public internet."

Comment (mmacy) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Also, line **678** (the "ports" property description) could do with a bit more explanatory text. Currently, it reads:

"The ports."

TCP only? UDP? Both? How do ports specified here relate to those on a container instance? Doesn't have to be long and involved, but a little more on this could go a long way.

**Review by mmacy — COMMENTED** [review]
This is looking way better! Just a few comments.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (yolo3301) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Volumes/image registry credentials are provided by the user when creating a container group. But IP is different. It's provided by ACI. Once a container group is deleted, the IP will be recycled. I think it would be more clear to say `The operation does not delete any resources provided by the users such as volumes`

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (yolo3301) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: It will follow the `docker run` [exit code](https://docs.docker.com/engine/reference/run/#exit-status). Maybe reference that?

**Comment by yolo3301** [comment]
@jluk I left a couple of comments. Mostly look good to me.

**Comment by jluk** [comment]
@yolo3301 made updates lmk thoughts. qq - what is the difference between port and containerPort?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Do we know when UDP will be supported, we can leave the documentation for it if it is very soon.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Is this accurate?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (mmacy) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: (very) Minor: "by the user, such as volumes." (add comma)

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (mmacy) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: How about:

"Exit codes correspond to those of the `docker run` command."

It feels a bit odd that that we're referencing an external tool in our own API reference documentation, but in this case, if it provides the best and most succinct explanation, it might be appropriate. Not sure there's another way to better communicate this, but wanted to call it out as a potential gotcha we might hit later on.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (mmacy) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: The enum value is "always," with lowercase 'a' -- should we maintain consistency with that? Also, a slightly modified option here, but might make it sound less like a limit, and perhaps even hint at future options:

"Restart policy for all containers within the container group. The currently available option is `always`."

**Review by mmacy — APPROVED** [review]
Looking great! Just a couple suggestions, but no blockers.

One thing I didn't comment on was ContainerPorts and possibly adding a bit of differentiation text between that and the container group port definition (can save this edit for a future quality pass).

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: @yolo3301 is there another way to reference the exit codes that can abstract us from docker?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: @yolo3301 from what I am reading in the swagger, "ports" are the ports utilized in the container group while "containerPorts" refer to individual instances. Could you verify my references are correct?

Comment (yolo3301) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: You're right. Ports under container group are the ports exposed with the public IP. Instance ports can be exposed for container-wise communication.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (yolo3301) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Maybe explicitly say about `tail`? Also if not provided, I think it will show up to 4mb logs. No more than that.

Comment (jluk) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Good feedback - added a bit about logs coming from the tail and the 4mb limit

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (wenwu449) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: "to be created or updated"

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (wenwu449) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: if there is no spec change, we also return 200, so 200 may be a no-op. Not sure if this information need to be exposed to customer.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (wenwu449) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: we only support delete on single container group, do we need the plural form groups?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Good point - removing additional messages from 200 responses to be only "OK".

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json
Comment (jluk) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json]: Agree, updating the description. However why are all the operationIds plural (ContainerGroups_Delete)? I suppose the plural form is the default ARM resource name is why?

**Comment by jluk** [comment]
@yolo3301 please make a pass and sign-off for @veronicagg to merge if it looks good. thanks everyone for the quick edits!

**Review by yolocs — APPROVED** [review]

**Comment by yolo3301** [comment]
@veronicagg I have approved the PR. Please help merge. Thanks!

**Review by veronicagg — APPROVED** [review]

**Pull request merged**
